src/app/page.tsx

```diff
@@ -1,16 +1,37 @@
 "use client";
 
-import styles from "./page.module.css";
+import Header from "@/components/Header/Header";
+import styles from "./page.module.sass";
 import CartProvider from "@/context/CartContext";
 import { makeServer } from "@/mirage/mirage";
+import { useCallback, useState, useMemo } from "react";
+import Sidebar from "@/components/Sidebar/Sidebar";
 
 makeServer();
 
 export default function Home() {
+  const [showSidebar, setShowSidebar] = useState(false);
+
+  // memoize <Header /> component so that
+  // it does not re-render when sidebar
+  // is toggled
+  const toggleSidebar = useCallback(() => {
+    setShowSidebar((prevState) => !prevState);
+  }, []);
+
+  const memoizedHeader = useMemo(() => {
+    return <Header toggleSidebar={toggleSidebar} />;
+  }, [toggleSidebar]);
+
   return (
     <main className={styles.main}>
       <CartProvider>
-        <div></div>
+        {showSidebar && (
+          <div className={styles.overlay} onClick={toggleSidebar}></div>
+        )}
+        <Sidebar showSidebar={showSidebar} toggleSidebar={toggleSidebar} />
```

Comment:
I think it is possible to move this to the header component instead of adding it to the page. What do you think?